src/libraries/System.Net.Sockets/tests/FunctionalTests/Connect.cs

```diff
@@ -127,7 +127,8 @@ public async Task ConnectGetsCanceledByDispose()
             // We try this a couple of times to deal with a timing race: if the Dispose happens
             // before the operation is started, we won't see a SocketException.
             int msDelay = 100;
-            await RetryHelper.ExecuteAsync(async () =>
+            int retries = 10;
+            while (true)
```

Comment:
```C#
                catch (SocketException se)
                {
                    // On connection timeout, retry.
                    Assert.NotEqual(SocketError.TimedOut, se.SocketErrorCode);

                    localSocketError = se.SocketErrorCode;
                }
```

The test fails here on Windows. Should do some more research on `SocketError.TimedOut` cases, but I think we should keep retrying here.